rotkehlchen/db/drivers/server.py

```diff
@@ -144,57 +138,64 @@ def _handle_query(
             starting_reference = db_connection
 
-        # Call the method on the starting reference with the specified arguments
-        return getattr(starting_reference, call_data.method.serialize())(*call_data.args, **call_data.kwargs)  # noqa: E501
+        return getattr(  # Call the method on the starting reference with the specified arguments
+            starting_reference,
+            DBMethod.Name(call_data.method).lower(),
+        )(*args, **kwargs)
 
     def listen(self) -> None:
         while True:
             try:
-                call_data: ZMQCallData = self.zmq_connection.recv_pyobj()
-                if call_data.method.value == DbMethod.INITIALIZE.value:
-                    self._handle_connection(call_data)
-                    self.zmq_connection.send_pyobj(None)
+                call_data_serialized = self.zmq_connection.recv()
+                ZMQReturnData.Clear()
+                ZMQCallData.ParseFromString(call_data_serialized)
+                args = tuple(from_typed_data(data) for data in ZMQCallData.args)
+                kwargs = {key: from_typed_data(data) for key, data in ZMQCallData.kwargs.items()}
+                if ZMQCallData.method == DBMethod.INITIALIZE:
+                    self._handle_connection(ZMQCallData, args, kwargs)
+                    self.zmq_connection.send(b'')
                     continue
 
                 try:
-                    db_connection = self.db_connections[call_data.db_path]
-                except KeyError:
-                    self.zmq_connection.send_pyobj(ZMQReturnData(
-                        result=None,
-                        error=KeyError(f'No DB connection for {call_data.connection_type} found'),
-                    ).serialize())
-                    continue
-
-                if call_data.method.value == DbMethod.OPEN_CURSOR.value:
-                    if self.db_cursors[call_data.db_path].get(call_data.cursor_name or 'default') is not None:  # noqa: E501
-                        raise KeyError(f'Cursor for {call_data.db_path} already exists')
-                    self.db_cursors[call_data.db_path][call_data.cursor_name or 'default'] = db_connection.cursor()  # noqa: E501
-                    self.zmq_connection.send_pyobj(None)
+                    db_connection = self.db_connections[ZMQCallData.db_path]
+                except KeyError as e:
+                    raise KeyError(f'No DB connection for {DBConnectionType(ZMQCallData.connection_type)} found') from e  # noqa: E501
+
+                if ZMQCallData.method == DBMethod.OPEN_CURSOR:
+                    if self.db_cursors[ZMQCallData.db_path].get(ZMQCallData.cursor_name or 'default') is not None:  # noqa: E501
+                        raise KeyError(f'Cursor for {ZMQCallData.db_path} already exists')
+                    self.db_cursors[ZMQCallData.db_path][ZMQCallData.cursor_name or 'default'] = db_connection.cursor()  # noqa: E501
```

Comment:
about the error message you should include also the name of the cursor since it's the name what is being used